consensus-peer/backend/consensus.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright IBM Corp. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backend
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/transport"
+
+	"golang.org/x/net/context"
+
+	"github.com/hyperledger/fabric/consensus"
+	"github.com/hyperledger/fabric/consensus-peer/connection"
+	pb "github.com/hyperledger/fabric/protos"
+	"github.com/op/go-logging"
+)
+
+var logger = logging.MustGetLogger("backend")
+
+type Backend struct {
+	consensus consensus.Consenter
+	conn      *connection.Manager
+
+	lock  sync.Mutex
+	peers map[pb.PeerID]chan<- *pb.Message
+
+	self     *PeerInfo
+	peerInfo map[string]*PeerInfo
+}
+
+type consensusConn Backend
+
+type PeerInfo struct {
+	info connection.PeerInfo
+	id   pb.PeerID
+}
+
+type peerInfoSlice []*PeerInfo
+
+func (pi peerInfoSlice) Len() int {
+	return len(pi)
+}
+
+func (pi peerInfoSlice) Less(i, j int) bool {
+	return strings.Compare(pi[i].info.Fingerprint(), pi[j].info.Fingerprint()) == -1
+}
+
+func (pi peerInfoSlice) Swap(i, j int) {
+	pi[i], pi[j] = pi[j], pi[i]
+}
+
+func New(peers map[string][]byte, conn *connection.Manager) (*Backend, error) {
+	c := &Backend{
+		conn:     conn,
+		peers:    make(map[pb.PeerID]chan<- *pb.Message),
+		peerInfo: make(map[string]*PeerInfo),
+	}
+
+	var peerInfo []*PeerInfo
+	for addr, cert := range peers {
+		pi, err := connection.NewPeerInfo(addr, cert)
+		if err != nil {
+			return nil, err
+		}
+		cpi := &PeerInfo{info: pi}
+		if pi.Fingerprint() == conn.Self.Fingerprint() {
+			c.self = cpi
+		}
+		peerInfo = append(peerInfo, cpi)
+		c.peerInfo[pi.Fingerprint()] = cpi
+	}
+
+	sort.Sort(peerInfoSlice(peerInfo))
+	for i, pi := range peerInfo {
+		pi.id.Name = fmt.Sprintf("vp%d", i)
+		logger.Infof("replica %d: %s", i, pi.info.Fingerprint())
```

Comment:
Very minor issue, but we should probably decide about capitalization for our log messages.  My impression had been that log messages start with a capital letter, but have no strong feelings.  If we are consistent, it makes grepping through logs a little easier.